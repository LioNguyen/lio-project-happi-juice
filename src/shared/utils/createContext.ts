import type React from "react";
import { createContext as originCreateContext, useContext } from "react";

function createContext<T>(): [React.Provider<T>, () => T, React.Context<T>] {
  const context = originCreateContext<T>({} as T);

  return [context.Provider, () => useContext(context), context];
}

export { createContext };
