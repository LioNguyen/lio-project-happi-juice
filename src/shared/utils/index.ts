export * from './lazyLoad'
export * from './tailwind'
