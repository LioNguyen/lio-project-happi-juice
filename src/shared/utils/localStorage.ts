/**
 * Save data to localStorage
 * @param key - The key under which the data will be stored
 * @param value - The data to store, can be any serializable value
 */
function setLocalStorage<T>(key: string, value: T): void {
  try {
    const serializedValue =
      typeof value === 'string' ? value : JSON.stringify(value)

    localStorage.setItem(key, serializedValue)
  } catch (error) {
    console.error(`Error saving ${key} to localStorage:`, error)
  }
}

/**
 * Retrieve data from localStorage
 * @param key - The key of the stored data
 * @returns The parsed data or null if not found or error occurs
 */
function getLocalStorage<T>(key: string): T | null {
  try {
    const serializedValue = localStorage.getItem(key)

    if (!serializedValue || serializedValue === 'undefined') {
      return null
    }

    // Try to parse JSON
    try {
      return JSON.parse(serializedValue) as T
    } catch {
      // Return as string if JSON parsing fails
      return serializedValue as T
    }
  } catch (error) {
    console.error(`Error reading ${key} from localStorage:`, error)
    return null
  }
}

/**
 * Remove data from localStorage
 * @param key - The key to remove from localStorage
 */
function removeLocalStorage(key: string): void {
  try {
    localStorage.removeItem(key)
  } catch (error) {
    console.error(`Error removing ${key} from localStorage:`, error)
  }
}

/**
 * Clear all data from localStorage
 */
function clearLocalStorage(): void {
  try {
    localStorage.clear()
  } catch (error) {
    console.error('Error clearing localStorage:', error)
  }
}

export {
  clearLocalStorage,
  getLocalStorage,
  removeLocalStorage,
  setLocalStorage,
}
