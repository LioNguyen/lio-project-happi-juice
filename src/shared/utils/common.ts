import { format, isValid, Locale, parseISO } from 'date-fns'
import { enUS, vi } from 'date-fns/locale'
import _ from 'lodash'

import { IOrderItem } from '@/domains/order'
import { ISelectOption } from '@/shared/types'

interface INumberFormatOptions {
  decimals?: number
  thousandSeparator?: string
  decimalSeparator?: string
}

// ==================== Date Utilities ====================
/**
 * Parse various date formats into Date object
 * @category Date
 * @param {Date | string | undefined} value - Input date
 * @returns {Date | undefined} Parsed Date object or undefined if invalid
 * @example
 * parseDate("2024-02-26") // Date object
 * parseDate("invalid") // undefined
 */
function parseDate(value: Date | string | undefined): Date | undefined {
  if (!value) return undefined
  if (value instanceof Date) return isValid(value) ? value : undefined

  try {
    // Try ISO string parsing first
    const date = parseISO(value)
    if (isValid(date)) return date

    // Fallback to direct Date creation
    const normalDate = new Date(value)
    return isValid(normalDate) ? normalDate : undefined
  } catch {
    return undefined
  }
}

/**
 * Format a date string according to specified format
 * @category Date
 * @param {string} date - Date string to format
 * @param {string} [dateFormat] - Output format (defaults to "yyyy-MM-dd")
 * @returns {string} Formatted date string
 * @example
 * formatDateTime("2024-02-26") // "2024-02-26"
 * formatDateTime("2024-02-26", "dd/MM/yyyy") // "26/02/2024"
 */
function formatDateTime(date: string, dateFormat?: string) {
  return format(date, dateFormat || 'yyyy-MM-dd')
}

const localeMap: Record<string, Locale> = {
  vi: vi,
  en: enUS,
}

// ==================== Number & Price Utilities ====================
/**
 * Format a number with customizable separators and decimals
 * @category Number
 * @param {number} value - Number to format
 * @param {INumberFormatOptions} [options] - Formatting options
 * @returns {string} Formatted number string
 */
const formatNumber = (
  value: number,
  options?: INumberFormatOptions,
): string => {
  const {
    decimals = 0,
    thousandSeparator = ',',
    decimalSeparator = '.',
  } = options || {}

  const parts = value.toFixed(decimals).split('.')
  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, thousandSeparator)

  return parts.join(decimalSeparator)
}

/**
 * Formats the price with thousand separators using locale-specific formatting
 * @category Price
 * @param {string} price - Price string to format
 * @returns {string} Formatted price string
 */
const formatPrice = (price: string) => {
  return Number(price).toLocaleString()
}

// ==================== Array Utilities ====================
/**
 * Transforms an array of select options into a sorted, joined string
 * @category Array
 * @param {ISelectOption[]} array - Array of select options
 * @param {string} [key] - Key to extract from each option
 * @param {string} [separator] - Separator for joining values
 * @returns {string} Sorted and joined string
 */
const transformToSortedString = (
  array: ISelectOption[],
  key: string = 'value',
  separator: string = ',',
): string => {
  return _.chain(array).map(key).sortBy().join(separator).value()
}

// ==================== Order Utilities ====================
/**
 * Calculates the total price for an array of order items
 * @category Order
 * @param {IOrderItem[]} items - Array of order items
 * @returns {number} Total price
 */
const calculateTotalPrice = (items: IOrderItem[]) => {
  return items.reduce((sum, item) => sum + item.price * item.quantity, 0)
}

export {
  // Date utils
  parseDate,
  formatDateTime,
  localeMap,

  // Number & Price utils
  formatNumber,
  formatPrice,

  // Array utils
  transformToSortedString,
  // Order utils
  calculateTotalPrice,
}
