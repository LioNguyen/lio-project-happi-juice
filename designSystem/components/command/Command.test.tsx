import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { vi } from "vitest";

import { renderWithTheme } from "@/shared/utils";
import {
  Command,
  CommandDialog,
  CommandEmpty,
  CommandGroup,
  CommandInput,
  CommandItem,
  CommandList,
  CommandSeparator,
  CommandShortcut,
} from "./Command";

// Mock CMDK
vi.mock("cmdk", () => {
  const CommandComponent = vi.fn(({ children, className, ...props }) => (
    <div data-testid="command" className={className} {...props}>
      {children}
    </div>
  ));

  return {
    Command: Object.assign(CommandComponent, {
      displayName: "Command",
      Input: vi.fn(({ className, ...props }) => (
        <div data-testid="command-input-wrapper" className="flex items-center border-b px-3">
          <input data-testid="command-input" className={className} {...props} />
        </div>
      )),
      List: vi.fn(({ className, children, ...props }) => (
        <div data-testid="command-list" className={className} {...props}>
          {children}
        </div>
      )),
      Empty: vi.fn(({ className, children, ...props }) => (
        <div data-testid="command-empty" className={className} {...props}>
          {children}
        </div>
      )),
      Group: vi.fn(({ className, children, ...props }) => (
        <div data-testid="command-group" className={className} {...props}>
          {children}
        </div>
      )),
      Separator: vi.fn(({ className, ...props }) => (
        <div data-testid="command-separator" className={className} {...props} />
      )),
      Item: vi.fn(({ className, children, ...props }) => (
        <div data-testid="command-item" className={className} {...props}>
          {children}
        </div>
      )),
    }),
  };
});

// Mock lucide-react
vi.mock("lucide-react", () => ({
  Search: vi.fn(() => <div data-testid="search-icon" />),
}));

// Mock Dialog components
vi.mock("@designSystem/components/dialog", () => ({
  Dialog: vi.fn(({ children, ...props }) => (
    <div data-testid="dialog" {...props}>
      {children}
    </div>
  )),
  DialogContent: vi.fn(({ children, className, ...props }) => (
    <div data-testid="dialog-content" className={className} {...props}>
      {children}
    </div>
  )),
}));

describe("Command Components", () => {
  describe("Command Snapshots", () => {
    it("basic command", () => {
      const { container } = renderWithTheme(
        <Command>
          <CommandInput placeholder="Type a command..." />
          <CommandList>
            <CommandEmpty>No results found.</CommandEmpty>
            <CommandGroup heading="Suggestions">
              <CommandItem>Calendar</CommandItem>
              <CommandItem>Search</CommandItem>
              <CommandItem>Settings</CommandItem>
            </CommandGroup>
            <CommandSeparator />
            <CommandGroup heading="Settings">
              <CommandItem>Profile</CommandItem>
              <CommandItem>Billing</CommandItem>
              <CommandItem>
                Settings
                <CommandShortcut>⌘S</CommandShortcut>
              </CommandItem>
            </CommandGroup>
          </CommandList>
        </Command>,
      );
      expect(container).toMatchSnapshot();
    });

    it("command dialog", () => {
      const { container } = renderWithTheme(
        <CommandDialog>
          <CommandInput placeholder="Type a command..." />
          <CommandList>
            <CommandEmpty>No results found.</CommandEmpty>
            <CommandGroup heading="Suggestions">
              <CommandItem>Command 1</CommandItem>
              <CommandItem>Command 2</CommandItem>
            </CommandGroup>
          </CommandList>
        </CommandDialog>,
      );
      expect(container).toMatchSnapshot();
    });

    it("command with shortcuts", () => {
      const { container } = renderWithTheme(
        <Command>
          <CommandList>
            <CommandGroup heading="Actions">
              <CommandItem>
                New File
                <CommandShortcut>⌘N</CommandShortcut>
              </CommandItem>
              <CommandItem>
                Save
                <CommandShortcut>⌘S</CommandShortcut>
              </CommandItem>
            </CommandGroup>
          </CommandList>
        </Command>,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Command", () => {
    it("should render with default props", () => {
      renderWithTheme(<Command />);

      const command = screen.getByTestId("command");
      expect(command).toBeInTheDocument();
      expect(command).toHaveClass(
        "flex",
        "h-full",
        "w-full",
        "flex-col",
        "overflow-hidden",
        "rounded-md",
        "bg-popover",
        "text-popover-foreground",
      );
    });

    it("should merge custom className", () => {
      renderWithTheme(<Command className="custom-class" />);

      const command = screen.getByTestId("command");
      expect(command).toHaveClass("custom-class");
    });
  });

  describe("CommandDialog", () => {
    it("should render dialog with command component", () => {
      renderWithTheme(
        <CommandDialog>
          <CommandInput placeholder="Search..." />
          <CommandList>
            <CommandEmpty>No results found</CommandEmpty>
          </CommandList>
        </CommandDialog>,
      );

      expect(screen.getByTestId("dialog")).toBeInTheDocument();
      expect(screen.getByTestId("dialog-content")).toHaveClass("overflow-hidden", "p-0");
      expect(screen.getByTestId("command")).toBeInTheDocument();
      expect(screen.getByPlaceholderText("Search...")).toBeInTheDocument();
    });
  });

  describe("CommandInput", () => {
    it("should render input with search icon", () => {
      renderWithTheme(<CommandInput placeholder="Search..." />);

      expect(screen.getByTestId("search-icon")).toBeInTheDocument();
      const inputWrapper = screen.getByTestId("command-input-wrapper");
      expect(inputWrapper).toHaveClass("flex", "items-center", "border-b", "px-3");

      const input = screen.getByPlaceholderText("Search...");
      expect(input).toHaveClass(
        "flex",
        "h-10",
        "w-full",
        "rounded-md",
        "bg-transparent",
        "py-3",
        "text-sm",
        "outline-none",
      );
    });

    it("should handle user input", async () => {
      const user = userEvent.setup();
      renderWithTheme(<CommandInput placeholder="Search..." />);

      const input = screen.getByPlaceholderText("Search...");
      await user.type(input, "test");
      expect(input).toHaveValue("test");
    });
  });

  describe("CommandList", () => {
    it("should render list with correct classes", () => {
      renderWithTheme(
        <CommandList>
          <CommandItem>Test Item</CommandItem>
        </CommandList>,
      );

      const list = screen.getByTestId("command-list");
      expect(list).toHaveClass("max-h-[300px]", "overflow-y-auto", "overflow-x-hidden");
      expect(screen.getByText("Test Item")).toBeInTheDocument();
    });
  });

  describe("CommandEmpty", () => {
    it("should render empty state", () => {
      renderWithTheme(<CommandEmpty>No results found</CommandEmpty>);

      const empty = screen.getByTestId("command-empty");
      expect(empty).toHaveClass("py-6", "text-center", "text-sm");
      expect(empty).toHaveTextContent("No results found");
    });
  });

  describe("CommandGroup", () => {
    it("should render group with heading", () => {
      renderWithTheme(
        <CommandGroup heading="Group Title">
          <CommandItem>Group Item</CommandItem>
        </CommandGroup>,
      );

      const group = screen.getByTestId("command-group");
      expect(group).toHaveClass("overflow-hidden", "p-1", "text-foreground");
      expect(screen.getByText("Group Item")).toBeInTheDocument();
    });
  });

  describe("CommandSeparator", () => {
    it("should render separator with correct styles", () => {
      renderWithTheme(<CommandSeparator />);

      const separator = screen.getByTestId("command-separator");
      expect(separator).toHaveClass("-mx-1", "h-px", "bg-border");
    });
  });

  describe("CommandItem", () => {
    it("should render item with correct styles", () => {
      renderWithTheme(<CommandItem>Item Text</CommandItem>);

      const item = screen.getByTestId("command-item");
      expect(item).toHaveClass(
        "relative",
        "flex",
        "cursor-default",
        "gap-2",
        "select-none",
        "items-center",
        "rounded-sm",
        "px-2",
        "py-1.5",
        "text-sm",
        "outline-none",
      );
    });

    it("should handle selected state", () => {
      renderWithTheme(<CommandItem data-selected={true}>Selected Item</CommandItem>);

      const item = screen.getByTestId("command-item");
      expect(item).toHaveClass("data-[selected=true]:bg-secondary", "data-[selected=true]:text-text-primary");
    });

    it("should handle disabled state", () => {
      renderWithTheme(<CommandItem data-disabled={true}>Disabled Item</CommandItem>);

      const item = screen.getByTestId("command-item");
      expect(item).toHaveClass("data-[disabled=true]:pointer-events-none", "data-[disabled=true]:opacity-50");
    });
  });

  describe("CommandShortcut", () => {
    it("should render shortcut with correct styles", () => {
      renderWithTheme(<CommandShortcut>⌘K</CommandShortcut>);

      const shortcut = screen.getByText("⌘K");
      expect(shortcut).toHaveClass("ml-auto", "text-xs", "tracking-widest", "text-text-muted-foreground");
    });
  });
});
