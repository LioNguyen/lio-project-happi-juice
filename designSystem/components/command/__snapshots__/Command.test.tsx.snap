// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Command Components > Command Snapshots > basic command 1`] = `
<div>
  <div
    class="flex h-full w-full flex-col overflow-hidden rounded-md bg-popover text-popover-foreground"
    data-testid="command"
  >
    <div
      class="flex items-center border-b px-3"
      cmdk-input-wrapper=""
    >
      <div
        data-testid="search-icon"
      />
      <div
        class="flex items-center border-b px-3"
        data-testid="command-input-wrapper"
      >
        <input
          class="flex h-10 w-full rounded-md bg-transparent py-3 text-sm outline-none placeholder:text-text-muted-foreground disabled:cursor-not-allowed disabled:opacity-50"
          data-testid="command-input"
          placeholder="Type a command..."
        />
      </div>
    </div>
    <div
      class="max-h-[300px] overflow-y-auto overflow-x-hidden custom-scrollbar"
      data-testid="command-list"
    >
      <div
        class="py-6 text-center text-sm"
        data-testid="command-empty"
      >
        No results found.
      </div>
      <div
        class="overflow-hidden p-1 text-foreground [&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:py-1.5 [&_[cmdk-group-heading]]:text-xs [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-text-muted-foreground"
        data-testid="command-group"
        heading="Suggestions"
      >
        <div
          class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
          data-testid="command-item"
        >
          Calendar
        </div>
        <div
          class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
          data-testid="command-item"
        >
          Search
        </div>
        <div
          class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
          data-testid="command-item"
        >
          Settings
        </div>
      </div>
      <div
        class="-mx-1 h-px bg-border"
        data-testid="command-separator"
      />
      <div
        class="overflow-hidden p-1 text-foreground [&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:py-1.5 [&_[cmdk-group-heading]]:text-xs [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-text-muted-foreground"
        data-testid="command-group"
        heading="Settings"
      >
        <div
          class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
          data-testid="command-item"
        >
          Profile
        </div>
        <div
          class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
          data-testid="command-item"
        >
          Billing
        </div>
        <div
          class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
          data-testid="command-item"
        >
          Settings
          <span
            class="ml-auto text-xs tracking-widest text-text-muted-foreground"
          >
            ⌘S
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Command Components > Command Snapshots > command dialog 1`] = `
<div>
  <div
    data-testid="dialog"
  >
    <div
      class="overflow-hidden p-0"
      data-testid="dialog-content"
    >
      <div
        class="flex h-full w-full flex-col overflow-hidden rounded-md bg-popover text-popover-foreground [&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-text-muted-foreground [&_[cmdk-group]:not([hidden])_~[cmdk-group]]:pt-0 [&_[cmdk-group]]:px-2 [&_[cmdk-input-wrapper]_svg]:h-5 [&_[cmdk-input-wrapper]_svg]:w-5 [&_[cmdk-input]]:h-12 [&_[cmdk-item]]:px-2 [&_[cmdk-item]]:py-3 [&_[cmdk-item]_svg]:h-5 [&_[cmdk-item]_svg]:w-5"
        data-testid="command"
      >
        <div
          class="flex items-center border-b px-3"
          cmdk-input-wrapper=""
        >
          <div
            data-testid="search-icon"
          />
          <div
            class="flex items-center border-b px-3"
            data-testid="command-input-wrapper"
          >
            <input
              class="flex h-10 w-full rounded-md bg-transparent py-3 text-sm outline-none placeholder:text-text-muted-foreground disabled:cursor-not-allowed disabled:opacity-50"
              data-testid="command-input"
              placeholder="Type a command..."
            />
          </div>
        </div>
        <div
          class="max-h-[300px] overflow-y-auto overflow-x-hidden custom-scrollbar"
          data-testid="command-list"
        >
          <div
            class="py-6 text-center text-sm"
            data-testid="command-empty"
          >
            No results found.
          </div>
          <div
            class="overflow-hidden p-1 text-foreground [&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:py-1.5 [&_[cmdk-group-heading]]:text-xs [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-text-muted-foreground"
            data-testid="command-group"
            heading="Suggestions"
          >
            <div
              class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
              data-testid="command-item"
            >
              Command 1
            </div>
            <div
              class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
              data-testid="command-item"
            >
              Command 2
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Command Components > Command Snapshots > command with shortcuts 1`] = `
<div>
  <div
    class="flex h-full w-full flex-col overflow-hidden rounded-md bg-popover text-popover-foreground"
    data-testid="command"
  >
    <div
      class="max-h-[300px] overflow-y-auto overflow-x-hidden custom-scrollbar"
      data-testid="command-list"
    >
      <div
        class="overflow-hidden p-1 text-foreground [&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:py-1.5 [&_[cmdk-group-heading]]:text-xs [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-text-muted-foreground"
        data-testid="command-group"
        heading="Actions"
      >
        <div
          class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
          data-testid="command-item"
        >
          New File
          <span
            class="ml-auto text-xs tracking-widest text-text-muted-foreground"
          >
            ⌘N
          </span>
        </div>
        <div
          class="relative flex cursor-default gap-2 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-secondary data-[selected=true]:text-text-primary data-[disabled=true]:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0"
          data-testid="command-item"
        >
          Save
          <span
            class="ml-auto text-xs tracking-widest text-text-muted-foreground"
          >
            ⌘S
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
